File: passenger-service/src/main/java/com/example/passenger_service/client/RatingServiceClient.java
package com.example.passenger_service.client;

import com.example.passenger_service.model.dto.PassengerRatingResponse;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;

import java.util.UUID;

@FeignClient(value = "rating-service")
public interface RatingServiceClient {
    @GetMapping("/api/v1/ratings/{targetUserId}")
    ResponseEntity<PassengerRatingResponse> getUserRating(@PathVariable UUID targetUserId);
}

File: driver-service/src/main/kotlin/com/example/driver_service/client/RatingServiceClient.kt
package com.example.driver_service.client

import com.example.driver_service.model.dto.DriverRatingResponse
import java.util.UUID
import org.springframework.cloud.openfeign.FeignClient
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable

@FeignClient(value = "rating-service")
interface RatingServiceClient {
    @GetMapping("/api/v1/ratings/{targetUserId}")
    fun getUserRating(@PathVariable targetUserId: UUID): ResponseEntity<DriverRatingResponse>
}